frontend/components/common/Table.tsx

```diff
@@ -0,0 +1,407 @@
+   }
+ },
+}
+
+type filterOption = {
+  options : string[],
+  label: string,
+}
+
+type Row= { [key: string]: any }
+
+type Action = {
+  name: string,
+  classes: string,
+  Icon : string,
+  clickFunction : (any) => any
+}
+
+type selectFilter = {
+  value: string,
+  label: string
+}
+ 
+type tableProps = {
+ columns: Row[],
+ data: Row[],
+ searchableColumns: string[]
+ filterOptions: filterOption [],
+ actions?: Action[]
+}
+ 
+const Styles = styled.div`
+ padding: 1rem;
+ width: 100%;
+ 
+ table {
+   border-collapse: collapse;
+   margin: 25px 0;
+   font-size: 0.95em;
+   font-family: sans-serif;
+   min-width: 400px;
+   box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
+ 
+   thead tr {
+     background-color: ${BLUE};
+     color: ${WHITE};
+     text-align: left;
+   }
+   thead th {
+    color: ${WHITE};
+    text-transform : capitalize
+  }
+ 
+   th,
+   td {
+     padding: 8px 15px;
+   }
+ 
+   tbody tr {
+     border-bottom: 1px solid #dddddd;
+   }
+ 
+   tbody tr:nth-of-type(even) {
+     background-color: ${HOVER_GRAY};
+   }
+ 
+   tbody tr:last-of-type {
+     border-bottom: 2px solid ${BLUE};
+   }
+
+ }
+`
+ 
+const SearchWrapper = styled.div`
+ overflow: visible;
+ margin: 0px;
+ ${mediaMaxWidth(MD)} {
+   width: 100%;
+   margin: 0;
+   border-bottom: 1px solid ${BORDER};
+   background: ${WHITE};
+ }
+`
+const Toolbar = styled.div`
+ padding-bottom: 25px;
+`
+const Input = styled.input`
+ border: 1px solid ${ALLBIRDS_GRAY};
+ outline: none;
+ color: ${CLUBS_GREY};
+ width: 100%;
+ font-size: 1em;
+ padding: 8px 10px;
+ background: ${WHITE};
+ border-radius: ${BORDER_RADIUS};
+ font-family: ${BODY_FONT};
+ 
+ &:hover,
+ &:active,
+ &:focus {
+   background: ${FOCUS_GRAY};
+ }
+`
+ 
+const Table = ({
+ columns,
+ data,
+ searchableColumns,
+ filterOptions,
+ actions,
+}: tableProps): ReactElement => {
+ const [searchQuery, setSearchQuery] = useState<string>('')
+ const [tableData, setTableData] = useState<Row[]>([])
+ const [selectedFilter, setSelectedFilter] = useState<selectFilter[]>([])
+ useEffect(() => {
+   const searchedData = data.filter((item) => {
+     if (!searchQuery || searchQuery.length < 0) return true
+     const searschString = searchQuery
+     if (!searschString || searschString.length < 3) {
+       return true
+     }
+     let valid = false
+     for (const searchId in searchableColumns) {
+       const strings = item[searchableColumns[searchId]].split(' ')
+       valid = strings.reduce(
+         (acc, item) =>
+           acc || item.toLowerCase().startsWith(searschString.toLowerCase()),
+         false,
+       )
+       if (
+         item[searchableColumns[searchId]]
+           .toLowerCase()
+           .startsWith(searschString.toLowerCase())
+       )
+         valid = true
+     }
+     return valid
+   })
+   const filteredData = searchedData.filter((item) => {
+     if (selectedFilter && selectedFilter.length > 0) {
+       let valid = true;
+       selectedFilter.forEach(filter => {
```

Comment:
You can replace this `forEach` loop with an [every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) statement.